### PR TITLE
"WIP resolved toLowerCase error currently  looking into ngIf error."

### PR DESCRIPTION
**Pull request by ocampd1**
Looking into making separate branch to work on ngIf error.

### CODE REVIEW

File: chucks-pick3-web/angular-app/src/app/components/demo.component/demo.component.ts [hunk 1]
```diff
@@ -7,10 +10,21 @@ import {Component, OnInit } from '@angular/core';
...
 
-  }
-}
+    if (this.route.snapshot.queryParamMap.get('showSubNavBar') === null) {
+      stop();
+
+
+    } else {
+      this.route.queryParamMap.pipe(map(params => {
+        return params.get('showSubNavBar').toLowerCase() === 'true' ? true : false;
```
Comment (pouncilt) [inline chucks-pick3-web/angular-app/src/app/components/demo.component/demo.component.ts]: Need to check if params is null or undefined first.  Only then can you call the `params.get('showSubNavBar')` method.  At which point this method call could return null or undefined.  You will need to do another check for that.  Only then can you call `.toLowerCase()` method.

Update this code and all other code in appropriate components.

File: chucks-pick3-web/angular-app/src/app/components/demo.component/demo.component.ts [hunk 2]
```diff
@@ -7,10 +10,21 @@ import {Component, OnInit } from '@angular/core';
...
+
```
Comment (pouncilt) [inline chucks-pick3-web/angular-app/src/app/components/demo.component/demo.component.ts]: remove empty line.

File: chucks-pick3-web/angular-app/src/app/components/info-generate.component/info-generate.component.ts [hunk 3]
```diff
@@ -15,12 +15,15 @@ export class InfoGenerateComponent implements OnInit {
...
-    // Capture the showSubNavBar if available
-    // tslint:disable-next-line:max-line-length
-    this.route.queryParamMap.pipe(map(params => params.get('showSubNavBar').toLowerCase() === 'true' ? true : false)).subscribe(subNavBarVisible => {
-      this.subNavBarService.setSubNavBarVisibility(subNavBarVisible);
-    });
-  }
 
-}
+    if (this.route.snapshot.queryParamMap.get('showSubNavBar') === null) {
+      stop();
```
Comment (pouncilt) [inline chucks-pick3-web/angular-app/src/app/components/info-generate.component/info-generate.component.ts]: What does this `stop()` method do?

Comment (ocampd1) [reply chucks-pick3-web/angular-app/src/app/components/info-generate.component/info-generate.component.ts]: The stop() method works in the same vain as a break within an if statement, thus preventing it from going into the last part of the function

File: chucks-pick3-web/angular-app/src/app/components/demo.component/demo.component.ts (hunk 1 above)
Comment (ocampd1) [reply chucks-pick3-web/angular-app/src/app/components/demo.component/demo.component.ts]: I've updated the code using 
 if (this.route.snapshot.queryParamMap.get('showSubNavBar') === null || undefined) {

I debugged it and did not run into issues.

Comment (ocampd1) [reply chucks-pick3-web/angular-app/src/app/components/demo.component/demo.component.ts]: I created validation checks for both the params and get(showSubNavBar).

File: chucks-pick3-web/angular-app/src/app/components/info-generate.component/info-generate.component.ts (hunk 3 above)
Comment (ocampd1) [reply chucks-pick3-web/angular-app/src/app/components/info-generate.component/info-generate.component.ts]: I found it within the DOM library but I'm replacing it with 'return'

Comment (pouncilt) [reply chucks-pick3-web/angular-app/src/app/components/info-generate.component/info-generate.component.ts]: Okay thanks.

**Pull request merged**